test/controllers/UpfrontPaymentControllerSpec.scala

```diff
@@ -43,121 +45,130 @@ class UpfrontPaymentControllerSpec extends ItSpec {
   private val expectedH1HowMuchCanYouPayUpfrontPage: String = "How much can you pay upfront?"
   private val expectedH1UpfrontSummaryPage: String = "Payment summary"
 
-  "GET /can-you-make-an-upfront-payment" - {
-
-    "should return 200 and the can you make an upfront payment page" in {
-      stubCommonActions()
-      EssttpBackend.EligibilityCheck.findJourney(testCrypto)()
-
-      val fakeRequest = FakeRequest().withAuthToken().withSession(SessionKeys.sessionId -> "IamATestSessionId")
-
-      val result: Future[Result] = controller.canYouMakeAnUpfrontPayment(fakeRequest)
-      val pageContent: String = contentAsString(result)
-      val doc: Document = Jsoup.parse(pageContent)
-
-      RequestAssertions.assertGetRequestOk(result)
-      ContentAssertions.commonPageChecks(
-        doc,
-        expectedH1              = expectedH1CanYouPayUpfrontPage,
-        shouldBackLinkBePresent = true,
-        expectedSubmitUrl       = Some(routes.UpfrontPaymentController.canYouMakeAnUpfrontPaymentSubmit.url)
-      )
-      val radioContent = doc.select(".govuk-radios__label").asScala.toList
-      radioContent(0).text() shouldBe "Yes"
-      radioContent(1).text() shouldBe "No"
-
-      doc.select("#CanYouMakeAnUpFrontPayment-hint").text() shouldBe expectedPageHintCanPayUpfrontPage
-    }
-
-    "should prepopulate the form when user navigates back and they have a chosen way to pay in their journey" in {
-      stubCommonActions()
-      EssttpBackend.CanPayUpfront.findJourney(testCrypto)()
+  Seq[(String, Origin, Enrolment, TaxRegime)](
+    ("EPAYE", Origins.Epaye.Bta, TdAll.payeEnrolment, TaxRegime.Epaye),
+    ("VAT", Origins.Vat.Bta, TdAll.vatEnrolment, TaxRegime.Vat)
+  ).foreach{
+      case (regime, origin, enrolment, taxRegime) =>
+        "GET /can-you-make-an-upfront-payment" - {
+
+          s"[$regime journey] should return 200 and the can you make an upfront payment page" in {
+            stubCommonActions(Some(Set(enrolment)))
```

Comment:
do the enrolments actually make a difference here? Don't the enrolments only really matter when looking for the tax id?

Reply:
seems to work ok when I change this for `stubCommonActions()` without the enrolments on my local machine. Saves a bit of code if it's not needed

Reply:
Ok will update